build-logic/packaging/src/main/kotlin/gradlebuild/instrumentation/tasks/InstrumentedMetadataMergeTask.kt

```diff
@@ -47,25 +52,40 @@ abstract class FindInstrumentedSuperTypesTask : DefaultTask() {
     @get:OutputFile
     abstract val instrumentedSuperTypesHash: RegularFileProperty
 
+    @get:OutputFile
+    abstract val upgradedProperties: RegularFileProperty
+
+    @get:OutputFile
+    abstract val upgradedPropertiesHash: RegularFileProperty
+
     @TaskAction
     fun run() {
         val instrumentedClasses = findInstrumentedClasses()
         if (instrumentedClasses.isEmpty()) {
             instrumentedSuperTypes.asFile.get().toEmptyFile()
             instrumentedSuperTypesHash.asFile.get().toEmptyFile()
-        } else {
-            // Merge and find all transitive super types
-            val superTypes = mergeSuperTypes()
-            // Keep only instrumented super types as we don't need to others for instrumentation
-            val onlyInstrumentedSuperTypes = keepOnlyInstrumentedSuperTypes(superTypes, instrumentedClasses)
-            writeOutput(onlyInstrumentedSuperTypes)
+            upgradedProperties.asFile.get().toEmptyFile()
+            upgradedPropertiesHash.asFile.get().toEmptyFile()
+            return
         }
+
+        mergeAndWriteInstrumentedSuperTypes(instrumentedClasses)
+        mergeAndWriteUpgradedProperties()
```

Comment:
🤔 You opted to add the new functionality to this existing task, and rename the task. To me the new functionality looks pretty much disjunct from the previously existing work. Would it make sense to add two separate tasks instead?

Reply:
Makes sense, it's splitted now